### ISSUE TITLE
Using spring-boot-starter-actuator in 1.2.0 with no security fails

### ISSUE THREAD

**Issue by jkirsch** [spring-projects/spring-boot#2142]
I upgraded from boot 1.1.9 to 1.2.0 and find that it's now not easily possible to run the actuators with no security without having spring security on the classpath

If I switch security off

`management.security.enabled=false`

I get the following error, of not finding `SessionCreationPolicy`

```
Caused by: java.lang.NoClassDefFoundError: org/springframework/security/config/http/SessionCreationPolicy
    at org.springframework.boot.actuate.autoconfigure.ManagementServerProperties$Security.<init>(ManagementServerProperties.java:142)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
    at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
    at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
    at java.lang.Class.newInstance(Class.java:374)
    at org.springframework.beans.BeanWrapperImpl.newValue(BeanWrapperImpl.java:649)
    at org.springframework.beans.BeanWrapperImpl.createDefaultPropertyValue(BeanWrapperImpl.java:622)
    at org.springframework.beans.BeanWrapperImpl.setDefaultValue(BeanWrapperImpl.java:610)
    at org.springframework.beans.BeanWrapperImpl.getNestedBeanWrapper(BeanWrapperImpl.java:574)
    at org.springframework.beans.BeanWrapperImpl.getBeanWrapperForPropertyPath(BeanWrapperImpl.java:548)
    at org.springframework.beans.BeanWrapperImpl.getPropertyDescriptorInternal(BeanWrapperImpl.java:358)
    at org.springframework.beans.BeanWrapperImpl.getPropertyType(BeanWrapperImpl.java:365)
    at org.springframework.boot.bind.RelaxedDataBinder.getActualPropertyName(RelaxedDataBinder.java:295)
    at org.springframework.boot.bind.RelaxedDataBinder.initializePath(RelaxedDataBinder.java:208)
    at org.springframework.boot.bind.RelaxedDataBinder.initializePath(RelaxedDataBinder.java:242)
    at org.springframework.boot.bind.RelaxedDataBinder.normalizePath(RelaxedDataBinder.java:200)
    at org.springframework.boot.bind.RelaxedDataBinder.modifyProperty(RelaxedDataBinder.java:181)
    at org.springframework.boot.bind.RelaxedDataBinder.modifyProperties(RelaxedDataBinder.java:145)
    at org.springframework.boot.bind.RelaxedDataBinder.doBind(RelaxedDataBinder.java:115)
    at org.springframework.validation.DataBinder.bind(DataBinder.java:630)
    at org.springframework.boot.bind.PropertiesConfigurationFactory.doBindPropertiesToTarget(PropertiesConfigurationFactory.java:251)
    at org.springframework.boot.bind.PropertiesConfigurationFactory.bindPropertiesToTarget(PropertiesConfigurationFactory.java:225)
    at org.springframework.boot.context.properties.ConfigurationPropertiesBindingPostProcessor.postProcessBeforeInitialization(ConfigurationPropertiesBindingPostProcessor.java:296)
    at org.springframework.boot.context.properties.ConfigurationPropertiesBindingPostProcessor.postProcessBeforeInitialization(ConfigurationPropertiesBindingPostProcessor.java:250)
    at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsBeforeInitialization(AbstractAutowireCapableBeanFactory.java:408)
    at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1558)
    at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:539)
    ... 32 more
```

If I now include spring security on the classpath, it works but I "secured" my whole application, so would need to reconfigure it.
I run the actuator on a different port in production which is hidden by the firewall so this not a real problem.

Is that something intended?

**Comment by guillaumegarcia13** [comment]
Using Spring Boot Actuator 1.4.1 and got this issue.
I had to add Spring Boot Security starter dependency to solve this.
Switching off security didn't seem to do the trick for me (not an expert in Spring, though)

**Comment by snicoll** [comment]
@guillaumegarcia13 this issue is fixed in the upcoming 1.4.2
